src/main/java/MetricCalculatorServlet.java

```diff
@@ -0,0 +1,52 @@
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+@WebServlet(name = "calculatorMetric", urlPatterns = "/calculatorMetric")
+public class MetricCalculatorServlet extends HttpServlet {
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        String meters = request.getParameter("meters");
+        String centimeters = request.getParameter("centimeters");
+        String milimeters = request.getParameter("milimeters");
+
+        response.setCharacterEncoding("UTF-8");
+        response.setContentType("text/html");
+        PrintWriter writer = response.getWriter();
+
+        if (!meters.isEmpty() && !centimeters.isEmpty() || !centimeters.isEmpty() && !milimeters.isEmpty() || !milimeters.isEmpty() && !meters.isEmpty()) {
```

Comment:
Zamiast pisać tak długie sprawdzenie, można to było zrobić na zasadzie metody która zlicza ile parametrów było nie pustych a następnie sprawdza czy ta liczba jest równa 1. Zwróci to `true` w przypadku kiedy wypełniony jest tylko jeden parametr, lub false jak więcej niż 1. A no i zapomniałeś o sprawdzeniu najpierw czy parametr nie jest `null` - może to wywołać niespodziewanie NPE